MekHQ/src/mekhq/AtBGameThread.java

```diff
@@ -317,33 +323,41 @@ public void run() {
-                            Set<Integer> toLoad = new HashSet<>();
-                            // Let the player choose to load fighters and/or ground units on each transport
-                            if (transport.isCarryingAero()) {
-                                loadFighters = (JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(null,
-                                                    String.format(AtBGameThread.LOAD_FTR_DIALOG_TEXT, transport.getName()),
-                                                    AtBGameThread.LOAD_FTR_DIALOG_TITLE, JOptionPane.YES_NO_OPTION));
-                            }
-                            if (transport.isCarryingGround()) {
-                                loadGround = (JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(null,
-                                        String.format(AtBGameThread.LOAD_GND_DIALOG_TEXT, transport.getName()),
-                                        AtBGameThread.LOAD_GND_DIALOG_TITLE, JOptionPane.YES_NO_OPTION));
-                            }
-                            // Now, send the load commands
-                            if (loadFighters || loadGround) {
-                                for (UUID cargoId : scenario.getPlayerTransportLinkages().get(id)) {
-                                    //Convert the list of Unit UUIDs to MM EntityIds
-                                    toLoad.add(campaign.getUnit(cargoId).getEntity().getId());
-                                }
-                                Utilities.loadPlayerTransports(transport.getEntity().getId(), toLoad, client, loadFighters, loadGround);
+                        Unit transport = campaign.getUnit(id);
+                        Set<Integer> toLoad = new HashSet<>();
+                        // Let the player choose to load fighters and/or ground units on each transport
+                        if (transport.isCarryingAero()) {
+                            loadFighters = (JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(null,
+                                                String.format(AtBGameThread.LOAD_FTR_DIALOG_TEXT, transport.getName()),
+                                                AtBGameThread.LOAD_FTR_DIALOG_TITLE, JOptionPane.YES_NO_OPTION));
+                        }
+                        if (transport.isCarryingGround()) {
+                            loadGround = (JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(null,
+                                    String.format(AtBGameThread.LOAD_GND_DIALOG_TEXT, transport.getName()),
+                                    AtBGameThread.LOAD_GND_DIALOG_TITLE, JOptionPane.YES_NO_OPTION));
+                        }
+                        // Now, send the load commands
+                        if (loadFighters || loadGround) {
+                            // List of technicians assigned to transported units. Several units can share a tech.
+                            Set<UUID> cargoTechs = new HashSet<>();
```

Comment:
I'm assuming this is to ensure the UUIDs are all unique?

Reply:
Correct.  You can have the same tech assigned to multiple units, and I only want one person to be added aboard the transport in such cases.